src/MessagePack.SourceGenerator/Analyzers/MsgPack00xMessagePackAnalyzer.cs

```diff
@@ -316,6 +328,46 @@ private void AnalyzeSymbol(SymbolAnalysisContext context, ReferenceSymbols typeR
             {
                 context.ReportDiagnostic(Diagnostic.Create(inaccessible, declaredSymbol.Locations[0]));
             }
+
+            // Call out any formattable reference types that are not nullable.
+            foreach (INamedTypeSymbol iface in declaredSymbol.Interfaces)
+            {
+                if (iface.TypeArguments is [INamedTypeSymbol { IsReferenceType: true, NullableAnnotation: NullableAnnotation.NotAnnotated } a])
+                {
+                    Location? location = null;
+
+                    // Find the location of the actual generic type argument if we can.
+                    foreach (var syntaxRef in declaredSymbol.DeclaringSyntaxReferences)
+                    {
+                        if (syntaxRef.GetSyntax(context.CancellationToken) is BaseTypeDeclarationSyntax { BaseList: { } baseList } typeSyntax)
+                        {
+#pragma warning disable RS1030 // Do not invoke Compilation.GetSemanticModel() method within a diagnostic analyzer
```

Comment:
Nit: maybe good to explain why it is OK to ignore this.

Reply:
I'm actually hoping the roslyn team will tell me what they recommend instead before I merge this.